main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintSolver2.scala

```diff
@@ -448,6 +455,11 @@ object ConstraintSolver2 {
       case other => Right(other)
     }
 
+    // Effect Provenance prototype
+    if (enableEffectProvenance) {
+      EffectProvenance.debug(eqConstrs)
```

Comment:
I don't think this is the right place to attach. I pointed out some other source location which is where an error occurs.